ding_dibs.email_reports.inc

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * @file
+ * E-mail reports functionality.
+ */
+
+
+/**
+ * Form settings mailing reports.
+ *
+ * @return
+ *   Save results form settings.
+ */
+function ding_dibs_setting_emails($form, &$form_state) {
+  drupal_add_css(drupal_get_path('module', 'ding_dibs') . '/css/ding_dibs.css');
+  $hour = range(0, 23);
+  $options_hour = array_map(_ding_dibs_generate_zero, $hour);
+  $minute = range(0, 59);
+  $options_minute = array_map(_ding_dibs_generate_zero, $minute);
+  $form = array();
+  $form['ding_dibs'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Set the daily hour to run this cron'),
+    '#prefix' => '<div class="container-inline">',
+    '#suffix' => '<div class="description">' . t('Set the daily hh:ii to run this cron.') . '</div></div>',
+  );
+  $form['ding_dibs']['ding_dibs_start_cron_hour'] = array(
+    '#type' => 'select',
+    '#options' => $options_hour,
+    '#default_value' => variable_get('ding_dibs_start_cron_hour', 2),
+    '#required' => TRUE,
+  );
+  $form['ding_dibs']['ding_dibs_start_cron_minute'] = array(
+    '#type' => 'select',
+    '#options' => $options_minute,
+    '#default_value' => variable_get('ding_dibs_start_cron_minute', 0),
+    '#required' => TRUE,
+  );
+  $form['ding_dibs_emails'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Input emails that should be notified if something unusual happens with a payment.'),
+    '#description' => t('Enter up to ten email addresses Enter through the separator or comma, without blank lines.'),
+    '#default_value' => variable_get('ding_dibs_emails', ''),
+    '#required' => TRUE,
+    '#element_validate' => array('ding_dibs_emails_validate'),
+  );
+  return system_settings_form($form);
+}
+
+/**
+ * Validation forms element email report settings.
+ */
+function ding_dibs_emails_validate($element, &$form_state, $form) {
+  $arrays_email = preg_split("/\,|\r\n|\n/", $element['#value']);
+  if (!empty($arrays_email)) {
+    foreach ($arrays_email as $email) {
+      $email = trim($email);
+      if (!empty($email) && !valid_email_address($email)) {
+        form_error($element, t('Please enter a valid email address.'));
+      }
+    }
+  }
+}
+
+/**
+ * Callback builds a table of reports to email mailing.
```

Comment:
Please revisit this DocBlock. Neither of us are native english speakers but this makes no sense.